### PR TITLE
Use the last chat message's created_at date as the "last message" date

### PR DESCRIPTION
**Pull request by SmittieC**

### CODE REVIEW

**Review by snopoke — APPROVED** [review]
The only issue here is that we're doing one query per item in the table. A more performant way to do it would be to modify the queryset by annotating it with the last message date (See https://github.com/dimagi/open-chat-studio/blob/92bd1b58c97e32bfe9c6ba68dfd09bdb7f5226ab/apps/events/models.py#L114C9-L159C10 for an example)

**Comment by SmittieC** [comment]
> The only issue here is that we're doing one query per item in the table. A more performant way to do it would be to modify the queryset by annotating it with the last message date (See https://github.com/dimagi/open-chat-studio/blob/92bd1b58c97e32bfe9c6ba68dfd09bdb7f5226ab/apps/events/models.py#L114C9-L159C10 for an example)

Good idea. See the latest commit

File: apps/experiments/views/experiment.py [hunk 1]
```diff
@@ -392,7 +399,16 @@ class DeleteFileFromExperiment(BaseDeleteFileView):
...
+    last_message_created_at = (
+        ChatMessage.objects.filter(
+            chat__experiment_session=OuterRef("pk"),
+        )
+        .order_by("-created_at")
+        .values("created_at")[:1]
+    )
+    user_sessions = ExperimentSession.objects.annotate(
+        last_message_created_at=Subquery(last_message_created_at)
+    ).filter(
```
Comment (snopoke) [inline apps/experiments/views/experiment.py:411]: nit: maybe worth making this a manager method to avoid the copy paste

**Review by snopoke — APPROVED** [review]

File: apps/experiments/views/experiment.py (hunk 1 above)
Comment (SmittieC) [reply apps/experiments/views/experiment.py:411]: Cool. Will throw this in with a followup PR. There are some other housekeeping things I want to do

**Pull request merged**